cmd/openfga/openfga_test.go

```diff
@@ -0,0 +1,967 @@
+	require.NoError(t, err)
+	defer conn.Close()
+
+	client := healthv1pb.NewHealthClient(conn)
+
+	backoffPolicy := backoff.NewExponentialBackOff()
+	backoffPolicy.MaxElapsedTime = 30 * time.Second
+
+	err = backoff.Retry(func() error {
+		timeoutCtx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
+		defer cancel()
+
+		resp, err := client.Check(timeoutCtx, &healthv1pb.HealthCheckRequest{
+			Service: openfgapb.OpenFGAService_ServiceDesc.ServiceName,
+		})
+		if err != nil {
+			return err
+		}
+
+		if resp.GetStatus() != healthv1pb.HealthCheckResponse_SERVING {
+			return fmt.Errorf("not serving")
+		}
+
+		return nil
+	},
+		backoffPolicy,
+	)
+	require.NoError(t, err)
+
+	return &serverHandle{
+		grpcPort: grpcPort,
+		httpPort: httpPort,
+		cleanup:  stopContainer,
+	}, nil
+}
+
+func TestFunctionalGRPC(t *testing.T) {
+
+	require := require.New(t)
+
+	// tester can be shared across tests that aren't impacted
+	// by shared state
+	tester, err := newOpenFGATester(t)
+	require.NoError(err)
+	defer tester.Cleanup()
+
+	t.Run("TestCreateStore", func(t *testing.T) { GRPCCreateStoreTest(t, tester) })
+	t.Run("TestGetStore", func(t *testing.T) { GRPCGetStoreTest(t, tester) })
+	t.Run("TestListStores", GRPCListStoresTest) // run an isolated tester from the others so bootstrapped stores don't collide
+	t.Run("TestDeleteStore", func(t *testing.T) { GRPCDeleteStoreTest(t, tester) })
+
+	t.Run("TestWrite", func(t *testing.T) { GRPCWriteTest(t, tester) })
+	t.Run("TestRead", func(t *testing.T) { GRPCReadTest(t, tester) })
+	t.Run("TestReadChanges", func(t *testing.T) { GRPCReadChangesTest(t, tester) })
+
+	t.Run("TestCheck", func(t *testing.T) { GRPCCheckTest(t, tester) })
+	t.Run("TestExpand", func(t *testing.T) { GRPCExpandTest(t, tester) })
+
+	t.Run("TestWriteAuthorizationModel", func(t *testing.T) { GRPCWriteAuthorizationModelTest(t, tester) })
+	t.Run("TestReadAuthorizationModel", func(t *testing.T) { GRPCReadAuthorizationModelTest(t, tester) })
+	t.Run("TestReadAuthorizationModels", func(t *testing.T) { GRPCReadAuthorizationModelsTest(t, tester) })
+}
+
+func TestGRPCWithPresharedKey(t *testing.T) {
+	tester, err := newOpenFGATester(t, "--authn-method", "preshared", "--authn-preshared-keys", "key1,key2")
+	require.NoError(t, err)
+	defer tester.Cleanup()
+
+	conn := connect(t, tester)
+
+	openfgaClient := openfgapb.NewOpenFGAServiceClient(conn)
+	healthClient := healthv1pb.NewHealthClient(conn)
+
+	resp, err := healthClient.Check(context.Background(), &healthv1pb.HealthCheckRequest{
+		Service: openfgapb.OpenFGAService_ServiceDesc.ServiceName,
+	})
+	require.NoError(t, err)
+	require.Equal(t, healthv1pb.HealthCheckResponse_SERVING, resp.Status)
+
+	_, err = openfgaClient.CreateStore(context.Background(), &openfgapb.CreateStoreRequest{
+		Name: "openfga-demo",
+	})
+	require.Error(t, err)
+
+	s, ok := status.FromError(err)
+	require.True(t, ok)
+	require.Equal(t, codes.Code(openfgapb.AuthErrorCode_bearer_token_missing), s.Code())
```

Comment:
This makes no sense to me as a client and as a maintainer of this project, haha 🤣 

What I should be asserting here is the following:

```
require.Equal(t, codes.Unauthenticated, s.Code())
```
We should adhere to the canonical status codes of the protocol we are using. It makes no sense for us to be implementing our own status code protocol ontop of a preexisting one.